diff --git a/src/pdm/core.py b/src/pdm/core.py
index e37cf9c643..f42d06be14 100644
--- a/src/pdm/core.py
+++ b/src/pdm/core.py
@@ -214,7 +214,14 @@ class Core:
         return -1, ""
 
     def _get_cli_args(self, args: list[str], obj: Project | None) -> list[str]:
-        project = self.create_project(is_global=False) if obj is None else obj
+        project_path = None
+        for index, argument in enumerate(args):
+            if argument == "-p" and index + 1 < len(args):
+                project_path = args[index + 1]
+                break
+        project = (
+            self.create_project(root_path=project_path, is_global=False) if obj is None else obj
+        )
         if project.is_global:
             return args
         try:
