diff --git a/src/tox/config/loader/ini/replace.py b/src/tox/config/loader/ini/replace.py
--- a/src/tox/config/loader/ini/replace.py
+++ b/src/tox/config/loader/ini/replace.py
@@ -41,6 +41,8 @@ class ReplaceReferenceIni(ReplaceReference):
                 for src in self._config_value_sources(settings["env"], settings["section"], conf_args.env_name):
                     try:
                         if isinstance(src, SectionProxy):
+                            if key in src:
+                                return ""
                             return self.loader.process_raw(self.conf, conf_args.env_name, src[key])
                         value = src.load(key, conf_args.chain)
                     except KeyError as exc:  # if fails, keep trying maybe another source can satisfy # noqa: PERF203
