diff --git a/loguru/_handler.py b/loguru/_handler.py
--- a/loguru/_handler.py
+++ b/loguru/_handler.py
@@ -204,6 +204,10 @@ class Handler:
                     self._queue.put(str_record)
                 else:
                     self._sink.write(str_record)
+        except KeyError:
+            if not self._error_interceptor.should_catch():
+                raise KeyError("Invalid logger format key")
+            self._error_interceptor.print(record, exception=KeyError("Invalid logger format key"))
         except Exception:
             if not self._error_interceptor.should_catch():
                 raise
