diff --git a/src/anyio/_backends/_asyncio.py b/src/anyio/_backends/_asyncio.py
--- a/src/anyio/_backends/_asyncio.py
+++ b/src/anyio/_backends/_asyncio.py
@@ -2322,5 +2322,10 @@ class TestRunner(abc.TestRunner):
             )
         except Exception as exc:
             self._exceptions.append(exc)
+        except BaseException:
+            if self._runner_task is not None and not self._runner_task.done():
+                self._runner_task.cancel()
+
+            raise
 
         self._raise_async_exceptions()
